### PR TITLE
Move strings in parser to resources issue #647

### PR DESCRIPTION
**Pull request by jamesralstin**
Moved hard-coded error messages from Parser.cs to Resources.resx. Ran unit tests from 'Microsoft.Pythong.Parsing.Tests.ParserTests', all tests passed except for 'Microsoft.Pythong.Parsing.Tests.ParserTests.Errors.' Noted failed test <a href="https://github.com/microsoft/python-language-server/issues/647#issuecomment-497893526">here</a>. Believe test failure existed prior to my changes, reverted failing code to original state, test still failed.

### CODE REVIEW

**Comment by MikhailArkhipov** [comment]
Yes, I cloned the branch and tests pass for me as well.

File: src/Parsing/Impl/Parser.cs [hunk 1]
```diff
@@ -2025,66 +2025,66 @@ struct Name {
...
                     }
                 } else if (!seenNames.Add(p.Name)) {
-                    ReportSyntaxError(p.StartIndex, p.EndIndex, "duplicate argument '{0}' in function definition".FormatUI(p.Name));
+                    ReportSyntaxError(p.StartIndex, p.EndIndex, Resources.DuplicateArgumentInFunctionErrorMsg.FormatUI(p.Name));
                 }
 
                 if (p.Kind == ParameterKind.List) {
                     if (seenListArg) {
-                        ReportSyntaxError(p.StartIndex, p.EndIndex, "duplicate * args arguments");
+                        ReportSyntaxError(p.StartIndex, p.EndIndex, Resources.DuplicateArgsArgumentErrorMsg.FormatInvariant("*"));//duplicate * args arguments
```
Comment (MikhailArkhipov) [inline src/Parsing/Impl/Parser.cs]: This probably does not need formatting, `*` can be part of the message as before.

File: src/Parsing/Impl/Parser.cs [hunk 2]
```diff
@@ -2025,66 +2025,66 @@ struct Name {
...
                     }
                     seenListArg = true;
                 } else if (p.Kind == ParameterKind.Dictionary) {
                     if (seenDictArg) {
-                        ReportSyntaxError(p.StartIndex, p.EndIndex, "duplicate ** args arguments");
+                        ReportSyntaxError(p.StartIndex, p.EndIndex, Resources.DuplicateArgsArgumentErrorMsg.FormatInvariant("**"));//duplicate ** args arguments
```
Comment (MikhailArkhipov) [inline src/Parsing/Impl/Parser.cs]: As above

File: src/Parsing/Impl/Parser.cs [hunk 3]
```diff
@@ -2025,66 +2025,66 @@ struct Name {
...
-                    ReportSyntaxError(p.StartIndex, p.EndIndex, "invalid syntax");
+                    ReportSyntaxError(p.StartIndex, p.EndIndex, Resources.InvalidSyntaxErrorMsg);//invalid syntax
                 }
             }
 
             if (parameters.Count > 0 && seenListArg) {
                 var p = parameters.Last();
                 if (p.Kind == ParameterKind.List && string.IsNullOrEmpty(p.Name)) {
-                    ReportSyntaxError(p.StartIndex, p.EndIndex, "named arguments must follow bare *");
+                    ReportSyntaxError(p.StartIndex, p.EndIndex, Resources.NamedArgumentsMustFollowBareErrorMsg);//named arguments must follow bare *
```
Comment (MikhailArkhipov) [inline src/Parsing/Impl/Parser.cs]: Ex, you don't format `*` here

File: src/Parsing/Impl/Parser.cs [hunk 4]
```diff
@@ -2161,7 +2161,7 @@ struct Name {
                 expr = ParseExpression(allowNamedExpressions: false);
             } else {
                 expr = Error(string.Empty);
-                ReportSyntaxError(_lookahead.Span.Start, _lookahead.Span.Start, "expected ':'");
+                ReportSyntaxError(_lookahead.Span.Start, _lookahead.Span.Start, Resources.ExpectedCharacterErrorMsg.FormatInvariant(":"));//"expected ':'"
```
Comment (MikhailArkhipov) [inline src/Parsing/Impl/Parser.cs]: No need to format as `:` is always the same here.

File: src/Parsing/Impl/Parser.cs [hunk 5]
```diff
@@ -4527,32 +4527,32 @@ struct Name {
...
                         }
                     } else if (hasArgsTuple || hasKeywordDict || keywordCount > 0) {
-                        ReportSyntaxError(arg.StartIndex, arg.EndIndex, "non-keyword arg after keyword arg");
+                        ReportSyntaxError(arg.StartIndex, arg.EndIndex, Resources.NonKeywordArgAfterKeywordArgErrorMsg);//non-keyword arg after keyword arg
                     }
                 } else if (arg.Name == "*") {
                     if (hasArgsTuple || hasKeywordDict) {
                         if (!_stubFile && _langVersion < PythonLanguageVersion.V35) {
-                            ReportSyntaxError(arg.StartIndex, arg.EndIndex, "only one * allowed");
+                            ReportSyntaxError(arg.StartIndex, arg.EndIndex, Resources.OnlyOneAllowedErrorMsg.FormatInvariant("*"));//"only one * allowed"
```
Comment (MikhailArkhipov) [inline src/Parsing/Impl/Parser.cs]: You can from formatting and include `*` in the resource as is

File: src/Parsing/Impl/Parser.cs [hunk 6]
```diff
@@ -4527,32 +4527,32 @@ struct Name {
...
-                            ReportSyntaxError(arg.StartIndex, arg.EndIndex, "iterable argument unpacking follows keyword argument unpacking");
+                            ReportSyntaxError(arg.StartIndex, arg.EndIndex, Resources.IterableArgumentFollowsKeywordArgumentErrorMsg);//iterable argument unpacking follows keyword argument unpacking
                         }
                     }
                     hasArgsTuple = true; extraArgs++;
                 } else if (arg.Name == "**") {
                     if (hasKeywordDict) {
                         if (!_stubFile && _langVersion < PythonLanguageVersion.V35) {
-                            ReportSyntaxError(arg.StartIndex, arg.EndIndex, "only one ** allowed");
+                            ReportSyntaxError(arg.StartIndex, arg.EndIndex, Resources.OnlyOneAllowedErrorMsg.FormatInvariant("**"));//only one ** allowed
```
Comment (MikhailArkhipov) [inline src/Parsing/Impl/Parser.cs]: As above

Comment (jamesralstin) [reply src/Parsing/Impl/Parser.cs]: Made recommended changes.

File: src/Parsing/Impl/Parser.cs (hunk 5 above)
Comment (jamesralstin) [reply src/Parsing/Impl/Parser.cs]: Made recommended changes.

File: src/Parsing/Impl/Parser.cs (hunk 4 above)
Comment (jamesralstin) [reply src/Parsing/Impl/Parser.cs]: Made recommended changes.

File: src/Parsing/Impl/Parser.cs (hunk 3 above)
Comment (jamesralstin) [reply src/Parsing/Impl/Parser.cs]: Made recommended changes.

File: src/Parsing/Impl/Parser.cs (hunk 2 above)
Comment (jamesralstin) [reply src/Parsing/Impl/Parser.cs]: Made recommended changes.

File: src/Parsing/Impl/Parser.cs (hunk 1 above)
Comment (jamesralstin) [reply src/Parsing/Impl/Parser.cs]: Made recommended changes.

**Comment by MikhailArkhipov** [comment]
Thanks! 👍

**Pull request merged**